studio/src/pages/RequestorPage/LogsTable/LogsTableRow/NeonEventRow.tsx

```diff
@@ -0,0 +1,134 @@
+import NeonLogo from "@/assets/NeonLogo.svg";
+import { CodeMirrorSqlEditor } from "@/components/CodeMirrorEditor";
+import { useFormattedNeonQuery } from "@/components/Timeline";
+import { Button } from "@/components/ui/button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { useCopyToClipboard } from "@/hooks";
+import { cn, getString, noop } from "@/utils";
+import { CopyIcon } from "@radix-ui/react-icons";
+import { useMemo, useState } from "react";
+import type { NeonEvent } from "../types";
+import { formatTimestamp } from "./shared";
+
+export function NeonEventRow({ log }: { log: NeonEvent }) {
+  // const textColor = "text-gray-500";
+  const [isExpanded, setIsExpanded] = useState(false);
+  // we don't want the focus ring to be visible when the user is selecting the row with the mouse
+  const [isMouseSelected, setIsMouseSelected] = useState(false);
+  const { isCopied: isMessageCopied, copyToClipboard: copyMessageToClipboard } =
+    useCopyToClipboard();
+
+  const isError = !!log.errors?.length;
+
+  const message = isError
+    ? "Neon DB Call failed"
+    : `Neon DB Call took ${log.duration}ms`;
+
+  const bgColor = isError ? "bg-red-500/10" : "bg-green-500/10";
+
+  const queryValue = useFormattedNeonQuery(log.sql);
+  // This will show `SELECT`, `INSERT`, etc. instead of the full query
+  const queryTypePreview = useMemo(() => {
+    const queryType = (queryValue.split("\n")?.[0] ?? "").toUpperCase();
+    return queryType.length < queryValue.length ? `${queryType}...` : queryType;
+  }, [queryValue]);
```

Comment:
I like the simple approach, but does this always work? When thinking about showing the type of the query i was thinking about using the response from neon (which contains the type of the SQL statement), but I guess this works assuming that drizzle always formats the query the same way (or perhaps it's not that bad if more than just `SELECT` is being returned here.

Reply:
so it works because of the sql formatter! but i like your solution better, didn't realize it was being returned by the neon api. fixed in latest commit!